sphinx/search/zh.py

```diff
@@ -234,7 +234,10 @@ def __init__(self, options: dict[str, str]) -> None:
 
     def init(self, options: dict[str, str]) -> None:
         if JIEBA:
-            dict_path = options.get('dict')
+            default_dict_path = os.path.join(
+                os.path.dirname(jieba.__file__), 'dict.txt'
```

Comment:
Can we use the jieba constant of `dict.txt` instead? or is this constant not meant to be publicly available?

If we want to only rely on public API and not on dunder attributes, we could open the file using a `with` statement, get the filename and then close it. It would probably be costly on Windows machines because open() calls can be up to 15x times slower compared to Linux but that would probably the second cleanest way to simplify maintenance.

Reply:
1. We can use `jieba.DEFAULT_DICT_NAME` to get the default dict name, but without its parent directory path.
2. We can use `inspect.getfile(jieba)` to get the path of the module `jieba`.
3. Finally we can use `os.path.join(os.path.dirname(inspect.getfile(jieba)), jieba.DEFAULT_DICT_NAME)` to get the path of the path of default dict file, which is not relying on dunder attributes.

One more thing,  I find the `inspect` provide a public API `inspect.getabsfile` which can get the absolute path of specify module, but it's likely not documented for a long year. cpython#56526

If not documented API is not a public API, it's better to use `inspect.getfile` instead, I think.

But at the same time, Github Copilot says that `inspect.getfile` may return a relative path, but I have not found any evdence on Internet.

Reply:
I have modify my code, please review again. @picnixz